### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/miri/actions/runs/4699927237#summary-12751183650

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/miri/actions/runs/4699927572#summary-12751184493

### CODE REVIEW

File: src/shims/os_str.rs [hunk 1]
```diff
@@ -329,7 +329,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
             match direction {
                 PathConversion::HostToTarget => {
                     // If this start withs a `\`, we add `\\?` so it starts with `\\?\` which is
-                    // some magic path on Windos that *is* considered absolute.
+                    // some magic path on Windows that *is* considered absolute.
```
Comment (jsoref) [inline src/shims/os_str.rs:332]: brand

File: tests/fail/unaligned_pointers/intptrcast_alignment_check.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 //@compile-flags: -Zmiri-symbolic-alignment-check -Zmiri-permissive-provenance -Cdebug-assertions=no
 // With the symbolic alignment check, even with intptrcast and without
 // validation, we want to be *sure* to catch bugs that arise from pointers being
-// insufficiently aligned. The only way to achieve that is not not let programs
+// insufficiently aligned. The only way to achieve that is not to let programs
```
Comment (jsoref) [inline tests/fail/unaligned_pointers/intptrcast_alignment_check.rs:4]: atypical correction

File: tests/pass/stacked-borrows/stacked-borrows.rs [hunk 3]
```diff
@@ -90,7 +90,7 @@ fn mut_raw_mut() {
         assert_eq!(unsafe { *xraw }, 4);
         assert_eq!(*xref1, 4);
         assert_eq!(unsafe { *xraw }, 4);
-        // we cannot use xref2; see `compile-fail/stacked-borows/illegal_read4.rs`
+        // we cannot use xref2; see `compile-fail/stacked-borrows/illegal_read4.rs`
```
Comment (jsoref) [inline tests/pass/stacked-borrows/stacked-borrows.rs:93]: this makes the comment match the path...

**Review by jsoref — COMMENTED** [review]
Most spelling corrections automatically suggested by Google Sheets (when being run on the rust repository). I've manually transplanted these commits into this repository.

All fault mine.

**Comment by RalfJung** [comment]
Thanks, the changes look good! However we don't need 52 commits of this, could you squash this into one?

**Comment by jsoref** [comment]
Sure, as long as you're happy with the changes. I might have time (and a laptop) this afternoon.

**Comment by saethlin** [comment]
I have also reviewed all the changes and I like them. I've been meaning to fix these at some point, just always had other things to worry about first.

**Comment by jsoref** [comment]
I should have asked if you had a preference for the commit message. This is my default, I'm not attached to it or anything else.

**Comment by RalfJung** [comment]
Seems fine. Thanks!
@bors r+

**Pull request merged**